test/examples/transactions.js

```diff
@@ -221,4 +222,69 @@ describe('examples(transactions):', function() {
       // End Transactions Retry Example 3
     }
   });
+
+  it('Transactions withTransaction API Example 1', {
+    metadata: { requires: { topology: ['replicaset'], mongodb: '>=3.8.0' } },
+    test: async function() {
+      const uri = this.configuration.url();
+
+      // Start Transactions withTxn API Example 1
+
+      // For a replica set, include the replica set name and a seedlist of the members in the URI string; e.g.
+      // const uri = 'mongodb://mongodb0.example.com:27017,mongodb1.example.com:27017/?replicaSet=myRepl'
+      // For a sharded cluster, connect to the mongos instances; e.g.
+      // const uri = 'mongodb://mongos0.example.com:27017,mongos1.example.com:27017/'
+
+      const client = new MongoClient(uri);
+      await client.connect();
+
+      // Prereq: Create collections. CRUD operations in transactions must be on existing collections.
+
+      client
+        .db('mydb1')
+        .collection('foo')
+        .insertOne({ abc: 0 }, { w: 'majority' });
+
+      client
+        .db('mydb2')
+        .collection('bar')
+        .insertOne({ xyz: 0 }, { w: 'majority' });
+
+      // Step 1: Start a Client Session
+
+      const session = client.startSession();
+
+      // Step 2: Optional. Define options to use for the transaction
+      const transactionOptions = {
+        readPreference: 'primary',
+        readConcern: { level: 'local' },
+        writeConcern: { w: 'majority' }
+      };
+
+      // Step 3: Use withTransaction to start a transaction, execute the callback, and commit (or abort on error)
+      // Note: The callback for withTransaction MUST be async and/or return a Promise.
+      let error;
+      try {
+        await session.withTransaction(async () => {
+          const coll1 = client.db('mydb1').collection('foo');
+          const coll2 = client.db('mydb2').collection('bar');
+
+          // Important:: You must pass the session to the operations
+
+          await coll1.insertOne({ abc: 1 }, { session });
+          await coll2.insertOne({ xyz: 999 }, { session });
+        }, transactionOptions);
+      } catch (e) {
```

Comment:
instead of storing the error in the `catch` and rethrowing after the try block, you can instead just put the `await client.close()` into a `finally` block. It will ensure that's called no matter what:
```
try {
  await session.withTransaction(...);
} finally {
  await client.close();
}

 => error is still thrown
```

Reply:
I omitted `try/finally` b/c it is only supported in Node 10. We currently run our examples in Node 8, so I'd have to change that if we want `try/finally`.

Reply:
wait I'm actually totally wrong about this. While `Promise.prototype.finally` does not exist in
Node 8, `try...finally` will still work? That is weird. I will change it.

Reply:
yeah it doesn't require `finally` on the promise, rather once you are using async/await you are just using the languages primitive try/catch/finally statements. Should work everywhere async/await is supported